addons/resource/static/tests/many2many_avatar_resource_tests.js

```diff
@@ -0,0 +1,186 @@
+            "/web/image/resource.resource/" + this.data.resource2Id + "/avatar_128",
+        );
+        assert.strictEqual(
+            document.querySelectorAll(".many2many_tags_avatar_field_container .o_tag img")[1].getAttribute("data-src"),
+            "/web/image/resource.resource/" + this.data.resource3Id + "/avatar_128",
+        );
+
+        // 1. Clicking on material resource's icon
+        await click(document.querySelector(".many2many_tags_avatar_field_container .o_tag i.fa-wrench"));
+        assert.containsNone(document.body, ".o_avatar_card");
+
+        // 2. Clicking on human resource's avatar with no user associated
+        await click(document.querySelector(".many2many_tags_avatar_field_container .o_tag img"));
+        await contains(".o_card_user_infos span", { text: "Marie" });
+        await contains(".o_avatar_card", { count: 1 }, 'Only one popover resource card should be opened at a time');
+        await contains(
+            ".o_avatar_card_buttons button",
+            { text: " Send message ", count: 0 },
+            'No "Send Message" button should be displayed for this employee as it is linked to no user',
+        );
+
+        // 3. Clicking on human resource's avatar with one user associated
+        await click(document.querySelectorAll(".many2many_tags_avatar_field_container .o_tag img")[1]);
+        await contains(".o_card_user_infos span", { text: "Pierre" });
+        await contains(".o_avatar_card", { count: 1 }, 'Only one popover resource card should be opened at a time');
+        await contains(".o_card_user_infos > a", { text: "Pierre@odoo.test" });
+        await contains(".o_card_user_infos > a", { text: "+32487898933" });
+        assert.strictEqual(document.querySelector(".o_avatar_card_buttons button").textContent, " Send message ");
+        await click(document.querySelector(".o_avatar_card_buttons button"));
+        await contains(".o-mail-ChatWindow");
+        assert.strictEqual(
+            document.querySelector('.o-mail-ChatWindow-header > .o-dropdown > button.o-mail-ChatWindow-command > .text-truncate').textContent,
+            "Pierre"
+        );
+    });
+
+    QUnit.test('many2many_avatar_resource widget in list view', async function (assert) {
+        this.serverData.views = {
+            "resource.task,false,list": `
+                    <tree>
+                        <field name="display_name"/>
+                        <field name="resource_ids" widget="many2many_avatar_resource"/>
+                    </tree>`,
+        };
+        const { openView } = await start({ serverData: this.serverData });
+        await openView({
+            res_model: "resource.task",
+            views: [[false, "list"]],
+        });
+
+        assert.containsN(document.body, '.o_m2m_avatar', 2, 'Two human resources with avatar should be displayed');
+        assert.containsN(document.body, 'i.fa-wrench', 1, 'Two material resources with fa-wrench icon should be displayed');
```

Comment:
```suggestion
        assert.containsN(document.body, ".o_m2m_avatar", 2, "Two human resources with avatar should be displayed");
        assert.containsN(document.body, "i.fa-wrench", 1, "Two material resources with fa-wrench icon should be displayed");
```
:slightly_frowning_face: